cores/arduino/WInterrupts.c

```diff
@@ -132,25 +160,37 @@ void detachInterrupt(uint32_t pin)
   
   // Disable wakeup capability on pin during sleep
   EIC->WAKEUP.reg &= ~(1 << in);
+
+  //and take it out of the ISR list
+  int id=-1; //find its location first
+  for (uint32_t i=0; i<nints; i++){
+      if (ISRlist[i]==in) id=in;
+  }
+  if (id==-1) return; //apparently we didn't have it
+  for (uint32_t i=id; i<nints-1; i++){ //and shift the ones above one down
+      ISRlist[i]=ISRlist[i+1];
+      ISRcallback[i]=ISRcallback[i+1];
+  }
+  //and remove the top item
+  ISRlist[nints]=0;
+  ISRcallback[nints]=NULL;
+  nints--;
 }
 
 /*
  * External Interrupt Controller NVIC Interrupt Handler
  */
 void EIC_Handler(void)
 {
-  // Test the 16 normal interrupts
-  for (uint32_t i=EXTERNAL_INT_0; i<=EXTERNAL_INT_15; i++)
-  {
-    if ((EIC->INTFLAG.reg & (1 << i)) != 0)
-    {
-      // Call the callback function if assigned
-      if (callbacksInt[i]) {
-        callbacksInt[i]();
-      }
-
-      // Clear the interrupt
-      EIC->INTFLAG.reg = 1 << i;
+  //calling the routine directly from -here- takes about 1us
+  //depending on where you are in the list it will take longer
+  for (uint32_t i=0; i<nints; i++) // Loop over all enabled interrupts in the list
+  {  
+    if ((EIC->INTFLAG.reg & 1<<ISRlist[i]) != 0)
+    {  
+      ISRcallback[i]();// Call the callback function if assigned
+      EIC->INTFLAG.reg = 1<<ISRlist[i]; // Clear the interrupt
```

Comment:
I'm wondering if we can gain something if we store `1 << ISRlist[i]` instead of `ISRlist[i]`.